spec/io/mat5_reader_spec.rb

```diff
@@ -0,0 +1,103 @@
+# = NMatrix
+#
+# A linear algebra library for scientific computation in Ruby.
+# NMatrix is part of SciRuby.
+#
+# NMatrix was originally inspired by and derived from NArray, by
+# Masahiro Tanaka: http://narray.rubyforge.org
+#
+# == Copyright Information
+#
+# SciRuby is Copyright (c) 2010 - 2014, Ruby Science Foundation
+# NMatrix is Copyright (c) 2012 - 2014, John Woods and the Ruby Science Foundation
+#
+# Please see LICENSE.txt for additional copyright notices.
+#
+# == Contributing
+#
+# By contributing source code to SciRuby, you agree to be bound by
+# our Contributor Agreement:
+#
+# * https://github.com/SciRuby/sciruby/wiki/Contributor-Agreement
+#
+# == fortran_format_spec.rb
+#
+# Basic tests for NMatrix::IO::Matlab
+#
+
+require_relative '../../lib/nmatrix'
+
+describe NMatrix::IO::Matlab do
+  def check_matrix_data_sparse arr
+    expect(arr[0][:nonzero_max]).to eq(11)
+    expect(arr[0][:matlab_class]).to eq(:mxSPARSE)
+    expect(arr[0][:dimensions]).to eq([4,4])
+    expect(arr[0][:matlab_name]).to eq("x")
+    expect(arr[0][:real_part][:tag][:data_type]).to eq(:miDOUBLE)
+    expect(arr[0][:real_part][:tag][:raw_data_type]).to eq(9)
+    expect(arr[0][:real_part][:tag][:bytes]).to eq(32)
+    expect(arr[0][:row_index][:tag][:data_type]).to eq(:miINT32)
+    expect(arr[0][:row_index][:tag][:raw_data_type]).to eq(5)
+    expect(arr[0][:row_index][:tag][:bytes]).to eq(20)
+    expect(arr[0][:column_index][:tag][:data_type]).to eq(:miINT32)
+    expect(arr[0][:column_index][:tag][:raw_data_type]).to eq(5)
+    expect(arr[0][:column_index][:tag][:bytes]).to eq(44)
+  end
+  def check_matrix_data_dense arr
+    expect(arr[0][:nonzero_max]).to eq(0)
+    expect(arr[0][:matlab_class]).to eq(:mxDOUBLE)
+    expect(arr[0][:dimensions]).to eq([4,5])
+    expect(arr[0][:matlab_name]).to eq("x")
+    expect(arr[0][:real_part][:tag][:data_type]).to eq(:miUINT8)
+    expect(arr[0][:real_part][:tag][:raw_data_type]).to eq(2)
+    expect(arr[0][:real_part][:tag][:bytes]).to eq(20)
+  end
+  it "reads MATLAB .mat file containing a single square sparse matrix" do
+    pending("not yet implemented for NMatrix-JRuby") if jruby?
+    n = NMatrix::IO::Matlab.load_mat("spec/4x4_sparse.mat")
```

Comment:
I have seen that you are using `n = NMatrix::IO::Matlab.load_mat(...)` in multiple test. Maybe you could use a `let` like:

```rb
let(:my_file) { "spec/4x4_sparse.mat" }
let(:n) { NMatrix::IO::Matlab.load_mat(my_file) }

it "reads MATLAB .mat file containing a single square sparse matrix" do
  expect(n[0,0]).to eq(2)
  ...
end
...
```
what do you think? 🤔